### ISSUE TITLE
Gitlab Flavored Markdown Annoying Formatting Error in Wiki

### ISSUE THREAD

**Issue by demisx** [gitlabhq/gitlabhq#7923]
If I use tripple backticks for formatting code as part of the ordered list, the code snippet is not formatted properly if it is in the last bullet and this is the only code snippet in the list (it just gets inserted highlighted inline):

1. Bullet 1
1. Bullet 2
1. Bullet 3
  ```bash
$ cd /tmp
$ git clone some_repo
```

I always need to follow it with a 4th bullet without the snippet to make it show up correctly, e.g.:

1. Bullet 1
1. Bullet 2
1. Bullet 3
  ```bash
$ cd /tmp
$ git clone some_repo
```

1. Bullet 4

 Very annoying and hard to remember when writing wiki pages and sometimes there is really no need to have this bullet 4. How can I make the last bullet in a list to properly show the indented code snippet?

**Comment by owenblacker** [comment]
There are also some other issues with the Markdown parsing.

If I have a code element inside an anchor:

> `` Like [this `example` here](http://example.com) perhaps``

I should expect output to look like

> Like [this `example` here](http://example.com) perhaps

I get output rendered as

> Like [this \<code\>example\</code\> here](http://example.com) perhaps

(This is using GitLab 7.7.2 98ef392 and the default parsing of a `README.md` file for a repo.)

**Comment by Razer6** [comment]
@owenblacker This is fixed in 7.12. See https://gitlab.com/test-repositories/link-to-tree/issues/1
